src/services/extension-registry-provider.js

```diff
@@ -112,7 +148,13 @@
                                           if(memo.length >= limit) {
                                             return memo;
                                           }
-                                          memo.push(next);
+                                          push(memo, next);
+                                          // - heaviest falls.
```

Comment:
True! that does read backwards.  I'll fix, or remove the comment since its in the README/Tests.